### PR TITLE
feat: add tick generation properties and legendColorizeRows

### PR DESCRIPTION
**Pull request by TCL735**
Follow on work to https://github.com/influxdata/giraffe/issues/79 and https://github.com/influxdata/giraffe/pull/317

Mirrors idpe pull 9027

Adds properties for storing the user's selections for tick generation, including a `generateAxisTicks` property that will turn on/off the entire feature regardless of the numeric values of the other properties.

Also adds the `legendColorizeRows` property to allow the user to toggle color on/off in the legend.

- [ ] [CHANGELOG.md](https://github.com/influxdata/influxdb/blob/master/CHANGELOG.md) updated with a link to the PR (not the Issue)
- [x] [Well-formatted commit messages](https://www.conventionalcommits.org/en/v1.0.0-beta.3/)
- [x] Rebased/mergeable
- [x] Tests pass
- [x] http/swagger.yml updated (if modified Go structs or API)
- [x] Feature flagged (if modified API)
- [x] Documentation updated or issue created (provide link to issue/pr)
- [x] Signed [CLA](https://influxdata.com/community/cla/) (if not already signed)

### CODE REVIEW

**Review by glinton — APPROVED** [review]

**Review by luismedicina23 — COMMENTED** [review]
Main problem 

Dashboard

**Review by luismedicina23 — COMMENTED** [review]
@TCL735 

Sry for merge

**Comment by TCL735** [comment]
> Main problem
> 
> Dashboard

Can you please say more? What is the problem?

**Review by luismedicina23 — COMMENTED** [review]
@TCL735

**Review by luismedicina23 — COMMENTED** [review]
@TCL735

**Review by luismedicina23 — COMMENTED** [review]
@TCL735 Sry
Im noob

File: dashboard_test.go [hunk 1]
```diff
@@ -48,12 +48,20 @@ func TestView_MarshalJSON(t *testing.T) {
     "geom": "",
     "note": "",
     "showNoteWhenEmpty": false,
-    "xColumn": "",
```
Comment (luismedicina23) [inline dashboard_test.go:51]: @main

Deleted line 51

File: dashboard_test.go [hunk 2]
```diff
@@ -48,12 +48,20 @@ func TestView_MarshalJSON(t *testing.T) {
...
-    "yColumn": "",
+		"generateAxisTicks": false,
+		"xColumn": "",
+		"xTotalTicks": 0,
+		"xTickStart": 0,
```
Comment (luismedicina23) [inline dashboard_test.go:54]:
```
		"xTickStart": 0,
```

**Review by luismedicina23 — COMMENTED** [review]
@Main

Mistake

Disaster

Plz I don't have chance to lern
@TCL735

**Pull request merged**